internal/storage/http.go

```diff
@@ -17,51 +14,8 @@ import (
 )
 
 type HTTP struct {
-	client      http.Client
-	requestOpts []func(*http.Request)
-}
-
-type HTTPOption func(*HTTP)
-
-func WithInsecureSkipVerify(v bool) HTTPOption {
-	return func(s *HTTP) {
-		tr := s.client.Transport.(*http.Transport)
-		if tr.TLSClientConfig == nil {
-			tr.TLSClientConfig = &tls.Config{}
-		}
-		tr.TLSClientConfig.InsecureSkipVerify = v
-	}
-}
-
-func WithRootCAs(rootCAs *x509.CertPool) HTTPOption {
-	return func(s *HTTP) {
-		tr := s.client.Transport.(*http.Transport)
-		if tr.TLSClientConfig == nil {
-			tr.TLSClientConfig = &tls.Config{}
-		}
-		tr.TLSClientConfig.RootCAs = rootCAs
-	}
-}
-
-func WithBearerToken(token string) HTTPOption {
-	return func(s *HTTP) {
-		s.requestOpts = append(s.requestOpts, func(request *http.Request) {
-			request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
-		})
-	}
-}
-
-type HTTPRequestOption func(*http.Request)
-
-func NewHTTP(opts ...HTTPOption) *HTTP {
-	s := &HTTP{client: http.Client{
-		Timeout:   time.Minute,
-		Transport: http.DefaultTransport.(*http.Transport).Clone(),
-	}}
-	for _, f := range opts {
-		f(s)
-	}
-	return s
+	Client      http.Client
+	RequestOpts []func(*http.Request)
```

Comment:
I'm not opposed to these changes but curious whether we could've kept these unexported if we had a constructor function that was responsible for setting them. You'd still be able to use the HTTP struct through the constructor, but the fields would be immutable after creating that resource.

Reply:
I tried but I couldn't create the constructor without causing the cycle dependencies.